x/concentrated-liquidity/math/math.go

```diff
@@ -18,8 +18,7 @@ func Liquidity0(amount osmomath.Int, sqrtPriceA, sqrtPriceB osmomath.BigDec) osm
 
 	// We convert to BigDec to avoid precision loss when calculating liquidity. Without doing this,
 	// our liquidity calculations will be off from our theoretical calculations within our tests.
-	// TODO (perf): consider better conversion helpers to minimize reallocations.
-	amountBigDec := osmomath.BigDecFromDec(amount.ToLegacyDec())
+	amountBigDec := osmomath.BigDecFromSDKInt(amount)
```

Comment:
Let's keep these comments please.

Do you mint creating an issue for this osmosis side and linking it to https://github.com/cosmos/cosmos-sdk/issues/17772?

Thanks!

Reply:
Sure!

Reply:
#6429